src/protocols/http.js

```diff
@@ -130,6 +132,15 @@ class HttpWrapper extends KuzzleAbstractProtocol {
         queryString.push(...value.map(v => `${key}=${v}`));
 
       }
+      else if (typeof value === 'boolean') {
+        // In Kuzzle, an optional boolean option is set to true if present in
+        // the querystring, and false if absent.
+        // As there is no boolean type in querystrings, encoding a boolean
+        // option "foo=false" in it will make Kuzzle consider it as truthy.
+        if (value === true) {
+          queryString.push(key);
+        }
```

Comment:
nitpicking: do we really need the comment block and the nested if?
```javascript
else if (typeof value === 'boolean' && value) {
  // foo=false => foo is truthy
  queryString.push(key);
}
```

Reply:
We need the nested if because, otherwise, a falsey boolen value will be processed by the final `else` condition just below, and added to the querystring anyway